package.json

```diff
@@ -79,6 +79,17 @@
         "icon": "$(graph-line)",
         "category": "Legend"
       },
+      {
+        "command": "legend.edit.service.query",
+        "title": "Edit Query",
+        "icon": "$(search)",
+        "category": "Legend"
+      },
+      {
+        "command": "legend.refresh.query.builder",
```

Comment:
What is a refresh of query builder?

Reply:
Refresh sends the latest entities to query builder. It's used to update the query builder UI if you save the service in the VS Code text editor.

Reply:
Wonder if we should automate these refreshes, including for diagrams?  We monitor file changes already, and we can create a notification mechanism to publish new entities when these are available...

Reply:
Yeah that makes sense! My only concern would be performance. Right now query builder rebuilds the whole graph when it gets net entities, but that could take a while in big projects. I'd like to investigate having 1 graph instance that's shared by all webviews and only rebuilding it when necessary.